analysis/train_value_function.py

```diff
@@ -23,7 +23,9 @@ def train_value_function(world, save_suffix="", scenario_training=True):
     for shift in range(number_of_shifts + 1):
         policy_world = copy.deepcopy(world)
         policy_world.policy.value_function.update_shifts_trained(shift)
-
+        policy_world.EPSILON = (
+            world.INITIAL_EPSILON - world.FINAL_EPSILON
+        ) / number_of_shifts
```

Comment:
Here you made a fast turn. Either you missed a minus sign before the equal sign or you wanted to divide by shift. Not seeing what you're trying to do, but what I can see from the PR, the Epsilon is constant for all shifts.